test/actions/pulp3/orchestration/file_sync_test.rb

```diff
@@ -40,5 +40,45 @@ def test_sync
 
       assert_equal repository_reference.repository_href + "versions/2/", @repo.version_href
     end
+
+    def test_sync_with_mirror_false
+      sync_args = {:smart_proxy_id => @master.id, :repo_id => @repo.id}
+      ForemanTasks.sync_task(::Actions::Pulp3::Orchestration::Repository::Sync, @repo, @master, sync_args)
+      @repo.reload
+      @repo.index_content
+      pre_count_content = ::Katello::RepositoryFile.where(:repository_id => @repo.id).count
+      @repo.root.update_attributes(:url => 'https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/file2/', :mirror_on_sync => false)
+
+      ForemanTasks.sync_task(
+          ::Actions::Pulp3::Orchestration::Repository::Update,
+          @repo,
+          @master)
+
+      ForemanTasks.sync_task(::Actions::Pulp3::Orchestration::Repository::Sync, @repo, @master, sync_args)
+      @repo.reload
+      @repo.index_content
+      post_count_content = ::Katello::RepositoryFile.where(:repository_id => @repo.id).count
+      assert_equal pre_count_content + 3, post_count_content
```

Comment:
I would probably prefer we don't rely on hosted repos whenever possible.  For yum its relatively easy as we can use the fixtures in ./test/fixtures/test_repos/  and add more repos there.  I'd probably rather do that then rely on pulp ones.  I can help you generate a yum repo there if you've not done it before.

Reply:
@jlsherrill : Does this https://github.com/Katello/katello/pull/8224/commits/cfbf4af092f8ed74d76acd19c611b142d5aa3076 look ok?

Reply:
Yep! looks great!